### ISSUE TITLE
PANIC: 'trust_cert' and 'trust_cert_ca' are mutual exclusive! Only use one.

### ISSUE THREAD

**Issue by mitchell-wallace** [prisma/prisma#14486]
Hi Prisma Team! My Prisma Client just crashed. This is the report:
## Versions

| Name            | Version            |
|-----------------|--------------------|
| Node            | v16.15.0           | 
| OS              | windows            |
| Prisma Client   | 4.1.0              |
| Query Engine    | 8d8414deb360336e4698a65aa45a1fbaf1ce13d8|
| Database        | sqlserver          |

## Logs
```
prisma:tryLoadEnv Environment variables loaded from C:\Users\mitch\Documents\GitHub\mudbath-sandbox\main\.env
prisma:tryLoadEnv Environment variables loaded from C:\Users\mitch\Documents\GitHub\mudbath-sandbox\main\.env
prisma:client dirname C:\Users\mitch\Documents\GitHub\mudbath-sandbox\main\node_modules\.prisma\client
prisma:client relativePath ..\..\..\prisma
prisma:client cwd C:\Users\mitch\Documents\GitHub\mudbath-sandbox\main\prisma
prisma:client clientVersion: 4.1.0
prisma:client clientEngineType: library
prisma:client:libraryEngine internalSetup
prisma:client:libraryEngine:loader Searching for Query Engine Library in C:\Users\mitch\Documents\GitHub\mudbath-sandbox\main\node_modules\.prisma\client
prisma:client:libraryEngine:loader loadEngine using C:\Users\mitch\Documents\GitHub\mudbath-sandbox\main\node_modules\.prisma\client\query_engine-windows.dll.node
prisma:client:libraryEngine sending request, this.libraryStarted: false
prisma:client:libraryEngine library starting
```

## Client Snippet
```ts
import pkg from '@prisma/client'
const { PrismaClient } = pkg;
const prisma = new PrismaClient()

import express from 'express'
const app = express();

async function main() {
    const user1 = await prisma.users.findUnique({
        where: {
        email: 'JaneDoe@example.com'
      },
        select: {
          email: true
        }
    }
    )
    console.log(user1)
```

## Schema
```prisma
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "sqlserver"
  url      = "sqlserver://localhost:1433;database=hashi_db;user=SA;password=myPassword;encrypt=true;trustServerCertificate=true;integratedSecurity=false;"
}

model users {=
  email        String  @id @db.VarChar(255)
  first_name   String? @db.VarChar(255)
  last_name    String? @db.VarChar(255)
  jira_id      String @db.VarChar(255)
  wrike_id     String @db.VarChar(255)
  created_by   String? @db.VarChar(255)
  created_date DateTime @default(now())
  updated_by   String? @db.VarChar(255)
  updated_date DateTime?
}

model projects {=
  project_name String  @id @db.VarChar(255)
  client_name  String? @db.VarChar(255)
  jira_id      String @db.VarChar(255)
  wrike_id     String @db.VarChar(255)
  created_by   String? @db.VarChar(255)
  created_date DateTime @default(now())
  updated_by   String? @db.VarChar(255)
  updated_date DateTime?
}
```

I have updated Prisma via npm today. Prisma studio is currently working correctly and is able to connect to my database.

**Comment by mitchell-wallace** [comment]
Some more info:

This is running in NodeJS on the local machine, connecting to a MSSQL database in Docker.

Here's the backtrace with RUST_BACKTRACE=full

```
thread 'tokio-runtime-worker' panicked at ''trust_cert' and 'trust_cert_ca' are mutual exclusive! Only use one.', C:\Users\runneradmin\.cargo\registry\src\github.com-1ecc6299db9ec823\tiberius-0.9.3\src\client\config.rs:151:13
stack backtrace:
   0:     0x7ff9f9fea87f - napi_register_module_v1
   1:     0x7ff9fa005eaa - napi_register_module_v1
   2:     0x7ff9f9fe3f29 - napi_register_module_v1
   3:     0x7ff9f9fed05b - napi_register_module_v1
   4:     0x7ff9f9fecc4e - napi_register_module_v1
   5:     0x7ff9f9fed651 - napi_register_module_v1
   6:     0x7ff9f9fed4d2 - napi_register_module_v1
   7:     0x7ff9f9feb187 - napi_register_module_v1
   8:     0x7ff9f9fed1e9 - napi_register_module_v1
   9:     0x7ff9fa19bab5 - napi_register_module_v1
  10:     0x7ff9f9d0da19 - napi_register_module_v1
  11:     0x7ff9f9d0e2ec - napi_register_module_v1
  12:     0x7ff9f9b528ac - napi_register_module_v1
  13:     0x7ff9f9ae1005 - napi_register_module_v1
  14:     0x7ff9f9414f5b - napi_register_module_v1
  15:     0x7ff9f9418b61 - napi_register_module_v1
  16:     0x7ff9f936ffb5 - napi_register_module_v1
  17:     0x7ff9f93565bb - napi_register_module_v1
  18:     0x7ff9f9363374 - napi_register_module_v1
  19:     0x7ff9f941f933 - napi_register_module_v1
  20:     0x7ff9f9443c91 - napi_register_module_v1
  21:     0x7ff9f9417ed9 - napi_register_module_v1
  22:     0x7ff9f8cb3f37 - <unknown>
  23:     0x7ff9f8cef504 - <unknown>
  24:     0x7ff9f8c88af2 - <unknown>
  25:     0x7ff9f8cf4f52 - <unknown>
  26:     0x7ff9f8da6106 - <unknown>
  27:     0x7ff9f8c3c00c - <unknown>
  28:     0x7ff9f8c406c6 - <unknown>
  29:     0x7ff9f9d25de1 - napi_register_module_v1
  30:     0x7ff9f9d37f84 - napi_register_module_v1
  31:     0x7ff9f9d36f02 - napi_register_module_v1
  32:     0x7ff9f9d15616 - napi_register_module_v1
  33:     0x7ff9f9d3678c - napi_register_module_v1
  34:     0x7ff9f9d19cdd - napi_register_module_v1
  35:     0x7ff9f9d1ddaf - napi_register_module_v1
  36:     0x7ff9f9d23754 - napi_register_module_v1
  37:     0x7ff9f9d3d36e - napi_register_module_v1
  38:     0x7ff9f9d2f3c8 - napi_register_module_v1
  39:     0x7ff9f9d2fd2b - napi_register_module_v1
  40:     0x7ff9f9ff161c - napi_register_module_v1
  41:     0x7ffa9959244d - BaseThreadInitThunk
  42:     0x7ffa9b5edf88 - RtlUserThreadStart
```

**Comment by garrensmith** [comment]
@pimeys could this be a problem in Tiberius?

**Comment by mitchell-wallace** [comment]
I downgraded to Prisma v3 and all is working properly now. Next week I should have some time to check through the major version update docs to see if there's anything I missed there.

**Comment by jkomyno** [comment]
This issue has also been reported [here](https://prisma-errors.netlify.app/report/14176).
@garrensmith, take this with a grain of salt, but it seems indeed related to [Tiberius](https://github.com/prisma/tiberius/blob/main/src/client/config.rs#L123-L155).

**Comment by garrensmith** [comment]
@mitchell-wallace looking at this again. This error seems correct. Can you check if you are accidentally setting both `trust_cert` and `trust_cert_ca`?

**Comment by pimeys** [comment]
Cannot replicate this either with 4.1.0 or 4.3.1. Is the connection string given in the issue the one that triggers the error? Could you maybe give us a reproduction in a repository we can validate against?

**Comment by janpio** [comment]
Ping @mitchell-wallace - can you help us out here? We rely on more information to be able to reproduce this, otherwise we will have to close the issue and wait until someone else reports it again. Thanks.

**Comment by mitchell-wallace** [comment]
Hi, sorry the project I was working on using Prisma had to change to a different tech stack for reasons unrelated to this issue. I'll try to find time to share a reproduction in the next week or so if possible.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by janpio** [comment]
Did you ever get to that @mitchell-wallace? We need your help to be able to reproduce this. Thanks!

**Comment by mitchell-wallace** [comment]
Sorry for the slow response.

A little story of how I found this issue. I had been connecting to a local MSSQL database, but had to change to connecting to one in docker, and it wasn't working at first so I was just trying anything to get it working. I found some flags to put in my connection string to trust the certificate, and to trust the CA, and set both of them to true, but it didn't change what errors I was getting. This was on Prisma 3.13. I then updated to Prisma 4.1 as a troubleshooting step and started getting this error of setting both `trust_cert` and `trust_cert_ca`, so I removed the flag to trust the CA, but I kept getting that error. I suspected that something outside the connection string was remembering these settings. I don't remember if at the time the connection string was directly in schema.prisma or in my .env file.

Now I've pulled up an old copy of this project and got it running. I can't find any documentation of what flag I was using to trust the CA, and I've looked in old files, source control, and online. It was very similar to the `trustServerCertificate` flag; I think it was `trustServerCertificateCA`, but I can't find any proof of that. I've tried a few guesses at what it might have been, while upgrading/downgrading between Prisma 3.13 and Prisma 4.1, but can't seem to reproduce that error message even when intentionally trying to set both `trust_cert` and `trust_cert_ca` via the connection string. Annoyingly, the old project seems to be working fine now.
